sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects/objects.scala

```diff
@@ -279,7 +282,8 @@ case class StaticInvoke(
     inputTypes: Seq[AbstractDataType] = Nil,
     propagateNull: Boolean = true,
     returnNullable: Boolean = true,
-    isDeterministic: Boolean = true) extends InvokeLike {
+    isDeterministic: Boolean = true,
+    scalarFunction: Option[ScalarFunction[_]] = None) extends InvokeLike {
```

Comment:
It seems you just try to get the name of scalar function, shall we only add the name parameters ?

Reply:
We need both `name` and `canonicalName` to build `UserDefinedScalarFunc`. So we need to add two extra parameters. Wouldn't it be more intuitive to use `ScalarFunction`?

Reply:
Please refer https://github.com/apache/spark/pull/42612#discussion_r1302453275

Reply:
OK, updated based on the comments. Replaced the `scalarFunction` with `scalarFunctionName` and `scalarFunctionCanonicalName`